### PR TITLE
Quantity literals

### PR DESCRIPTION
**Pull request by plynchnlm**

### CODE REVIEW

File: src/fhirpath.js [hunk 1]
```diff
@@ -219,6 +220,18 @@ engine.BooleanLiteral = function(ctx, parentData, node) {
...
 };
 
+engine.QuantityLiteral = function(ctx, parentData, node) {
+  var valueNode = node.children[0];
+  var value = Number(valueNode.terminalNodeText[0]);
+  var unitNode = valueNode.children[0];
+  var unit = unitNode.terminalNodeText[0];
+  // Sometimes the unit is in a child node of the child
+  if (!unit && unitNode.children)
+    unit = unitNode.children[0].terminalNodeText[0];
```
[the remark this answers is not in the document]
Comment (plynchnlm) [reply src/fhirpath.js]: The parser (generated by ANTLR) controls the structure.  The difference in structure is between the UCUM units and the non-UCUM units (which I think are handled differently by the grammar).

File: src/types.js [hunk 2]
```diff
@@ -49,6 +49,16 @@ class FP_Type {
 }
 
 
+class FP_Quantity extends FP_Type {
+  constructor(value, unit) {
+    super();
+    this.asStr = value + ' ' + unit;
```
[the remark this answers is not in the document]
Comment (plynchnlm) [reply src/types.js]: UCUM units are single-quoted, and the "unit" variable would already contain the single-quotes in its string.

File: test/fhirpath.test.js [hunk 3]
```diff
@@ -55,7 +55,9 @@ const generateTest = (test, testResource) => {
     }
     if (!test.error && test.expression) {
       const result = calcExpression(expression, test, testResource);
-      expect(result).toEqual(test.result);
+      // Run the result through JSON so the FP_Type quantities get converted to
+      // strings.
+      expect(JSON.parse(JSON.stringify(result))).toEqual(test.result);
```
[the remark this answers is not in the document]
Comment (plynchnlm) [reply test/fhirpath.test.js]: The "expect" is operating on the objects, so there is no assumption of key order.

[the remark this answers is not in the document]
Comment (xluanlhc) [reply test/fhirpath.test.js]: Right, I read it the wrong direction (thinking was comparing the string)

**Pull request merged**